py/server/deephaven/experimental/s3.py

```diff
@@ -118,12 +145,15 @@ def __init__(self,
                     (access_key_id is None and secret_access_key is not None)):
                 raise DHError("Either both access_key_id and secret_access_key must be provided or neither")
 
+            # Only one of the following credentials can be set, otherwise the builder will throw an exception
             if access_key_id is not None:
-                if anonymous_access:
-                    raise DHError("Only one set of credentials may be used, requested both key and anonymous")
                 builder.credentials(_JCredentials.basic(access_key_id, secret_access_key))
-            elif anonymous_access:
+            if anonymous_access:
                 builder.credentials(_JCredentials.anonymous())
+            if profile_credentials:
+                builder.credentials(_JCredentials.profile())
+            if default_credentials:
+                builder.credentials(_JCredentials.defaultCredentials())
```

Comment:
Docs say that this can not be specified with other combinations.  I do not see a check for that condition in python.  Does the check happen in java?  Is the resulting exception useful in Python?

Reply:
Replied in the comment above.